vm/src/protocol/object.rs

```diff
@@ -317,8 +317,12 @@ impl PyObject {
 
     pub fn repr(&self, vm: &VirtualMachine) -> PyResult<PyStrRef> {
         vm.with_recursion("while getting the repr of an object", || {
-            let repr = vm.call_special_method(self.to_owned(), identifier!(vm, __repr__), ())?;
-            repr.try_into_value(vm)
+            match self.class().slots.repr.load() {
+                Some(slot) => slot(self, vm),
+                None => vm
+                    .call_special_method(self.to_owned(), identifier!(vm, __repr__), ())?
```

Comment:
If builtin types fully implemented `__repr__` through `Representable` trait, this is not necessary. `slot` always exists and it will call `__repr__`.

Reply:
Yes I guess I just need to continue my implementation work for all the stdlib as well 😅

Reply:
Oh, please don't do it now. I thought it is fully transited. but it is ok to keep it until we finish to migrate them later.

Reply:
I thought I also needed that `vm.call_special_method` for pure python class but it turns out I really only need it for the stdlib so I'll finish cleanly that PR 🙂 There's only 24 methods left to patch so it is doable.